src/pal/pal_apple.h

```diff
@@ -18,75 +28,158 @@ namespace snmalloc
   public:
     /**
      * The features exported by this PAL.
-     *
-     * Currently, these are identical to the generic BSD PAL.  This field is
-     * declared explicitly to remind anyone who modifies this class that they
-     * should add any required features.
      */
-    static constexpr uint64_t pal_features = PALBSD::pal_features;
+    static constexpr uint64_t pal_features =
+      AlignedAllocation | LazyCommit | Entropy;
+
+    static constexpr size_t page_size = Aal::aal_name == ARM ? 0x4000 : 0x1000;
+
+    static constexpr size_t minimum_alloc_size = page_size;
 
-    /**
-     * Anonymous page tag ID.
-     *
-     * Darwin platform allows to gives an ID to anonymous pages via
-     * the VM_MAKE_TAG's macro, from 240 up to 255 are guaranteed
-     * to be free of usage, however eventually a lower could be taken
-     * (e.g. LLVM sanitizers has 99) so we can monitor their states
-     * via vmmap for instance. This value is provided to `mmap` as the file
-     * descriptor for the mapping.
-     */
     static constexpr int anonymous_memory_fd = VM_MAKE_TAG(PALAnonID);
 
-    /**
-     * Note: The root's implementation works fine on Intel
-     * however mprotect/PROT_NONE fails on ARM
-     * especially since the 11.2 release (seems known issue
-     * spotted in various projects; might be a temporary fix).
-     */
-    template<bool page_aligned = false>
-    static void zero(void* p, size_t size) noexcept
+    static constexpr int default_mach_vm_map_flags = VM_MAKE_TAG(PALAnonID);
+
+    static void print_stack_trace()
     {
-      if constexpr (Aal::aal_name != ARM)
-        PALBSD::zero(p, size);
-      else
-        ::bzero(p, size);
+      constexpr int SIZE = 1024;
+      void* buffer[SIZE];
+      auto nptrs = backtrace(buffer, SIZE);
```

Comment:
curiosity question, does it bring useful info ? backtrace on Linux is usually good however I have mostly lower experiences with other oses.

Reply:
I switched back the PALPOSIX's definition.

Reply:
glad you did